src/Symfony/Component/HttpKernel/BootingKernel.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel;
+
+use Symfony\Component\Config\Loader\LoaderInterface;
+use Symfony\Component\DependencyInjection\Exception\BadMethodCallException;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Bundle\BootingBundle;
+use Symfony\Component\HttpKernel\Bundle\BundleInterface;
+
+/**
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+class BootingKernel implements KernelInterface
+{
+    private $innerKernel;
+
+    public function __construct(KernelInterface $innerKernel)
+    {
+        $this->innerKernel = $innerKernel;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function serialize()
+    {
+        return $this->innerKernel->serialize();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function unserialize($serialized)
+    {
+        return $this->innerKernel->unserialize($serialized);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(Request $request, $type = self::MASTER_REQUEST, $catch = true)
+    {
+        throw new BadMethodCallException(sprintf('Calling "%s()" is not allowed.', __METHOD__));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function registerBundles()
+    {
+        throw new BadMethodCallException(sprintf('Calling "%s()" is not allowed.', __METHOD__));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function registerContainerConfiguration(LoaderInterface $loader)
+    {
+        throw new BadMethodCallException(sprintf('Calling "%s()" is not allowed.', __METHOD__));
```

Comment:
looking at the code, it's actually done that way everywhere so I'm actually not so sure about that after all... IMO to be easier to maintain, you would be better off having a `BadMethodCallExceptionFactory::create(string $method)` which could be reused anywhere